trikot-viewmodels/viewmodels/src/androidMain/kotlin/com/mirego/trikot/viewmodels/lifecycle/ApplicationStatePublisher.kt

```diff
@@ -21,11 +22,17 @@ actual class ApplicationStatePublisher :
 
     override fun onFirstSubscription() {
         super.onFirstSubscription()
-        lifecycle.addObserver(this)
+
+        UIThreadDispatchQueue().dispatch {
```

Comment:
We could reuse the same dispatch queue for both methods, but that's fine like that too 🙂